bayesian/utility/v2/ca_response_builder.py

```diff
@@ -159,26 +159,21 @@ def get_version_without_cves(self, latest_non_cve_versions):
 
         :return: Highest Version out of all latest_non_cve_versions.
         """
-        logger.debug("All Versions with Vulnerabilities.")
-        input_version_tuple = version_info_tuple(
-            convert_version_to_proper_semantic(self.version)
-        )
         highest_version = ''
-        for version in latest_non_cve_versions:
-
-            graph_version_tuple = version_info_tuple(
-                convert_version_to_proper_semantic(version)
-            )
-            if graph_version_tuple > input_version_tuple:
-                if not highest_version:
-                    highest_version = version
-                highest_version_tuple = version_info_tuple(
-                    convert_version_to_proper_semantic(highest_version)
-                )
-                # If version to recommend is closer to what a user is using then, use less than
-                # If recommendation is to show highest version then, use greater than
-                if graph_version_tuple > highest_version_tuple:
-                    highest_version = version
+        try:
+            input_version_comparable = ComparableVersion(self.version)
+
+            # latest non-cve is list with only one entry.
+            latest_non_cve_version_comparable = ComparableVersion(latest_non_cve_versions[0])
+
+            if latest_non_cve_version_comparable > input_version_comparable:
+                highest_version = latest_non_cve_versions[0]
```

Comment:
This must go out of exception else block.

Reply:
This  variables are declared and initialized within this block, its good to perform this one step within the block?

Reply:
you can still access those from the exception else block.

Reply:
Yes, your right. From interface perspective, compare_to() interface of `ComparableVersion` can throw TypeError(), So as a caller it better we handle that exception. Also I don't see a big advantage of moving this condition statement outside the try catch block.

Reply:
where is that described?

Reply:
If you go by documentation, even `ComparableVersion`  should also not throw any error as its not documented, so we don't need this try catch block itself. Let me move this out, just to make you happy.... I still feel its better to keep this under a try catch and does not add any overhead.